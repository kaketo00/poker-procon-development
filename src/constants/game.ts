import type { Card } from '@/schema/card';
import { Hand } from '@/schema/game';

export const MIN_PLAYER = 4; // 参加者数
export const HAND_SIZE = 5;
export const DEFAULT_ROUND = 100;
export const DEFAULT_INITIAL_POINT = 20000;
export const DEFAULT_FEE = 200;

export const GAME_STATUS = {
  new: {
    label: '新規',
    color: 'brand100.300',
  },
  progress: {
    label: '進行中',
    color: 'brand100.200',
  },
  finished: {
    label: '終了',
    color: 'brand100.600',
  },
};

export const CARDS: Card[] = [
  { suit: 'Spades', number: 1 },
  { suit: 'Spades', number: 2 },
  { suit: 'Spades', number: 3 },
  { suit: 'Spades', number: 4 },
  { suit: 'Spades', number: 5 },
  { suit: 'Spades', number: 6 },
  { suit: 'Spades', number: 7 },
  { suit: 'Spades', number: 8 },
  { suit: 'Spades', number: 9 },
  { suit: 'Spades', number: 10 },
  { suit: 'Spades', number: 11 },
  { suit: 'Spades', number: 12 },
  { suit: 'Spades', number: 13 },
  { suit: 'Clubs', number: 1 },
  { suit: 'Clubs', number: 2 },
  { suit: 'Clubs', number: 3 },
  { suit: 'Clubs', number: 4 },
  { suit: 'Clubs', number: 5 },
  { suit: 'Clubs', number: 6 },
  { suit: 'Clubs', number: 7 },
  { suit: 'Clubs', number: 8 },
  { suit: 'Clubs', number: 9 },
  { suit: 'Clubs', number: 10 },
  { suit: 'Clubs', number: 11 },
  { suit: 'Clubs', number: 12 },
  { suit: 'Clubs', number: 13 },
  { suit: 'Diamonds', number: 1 },
  { suit: 'Diamonds', number: 2 },
  { suit: 'Diamonds', number: 3 },
  { suit: 'Diamonds', number: 4 },
  { suit: 'Diamonds', number: 5 },
  { suit: 'Diamonds', number: 6 },
  { suit: 'Diamonds', number: 7 },
  { suit: 'Diamonds', number: 8 },
  { suit: 'Diamonds', number: 9 },
  { suit: 'Diamonds', number: 10 },
  { suit: 'Diamonds', number: 11 },
  { suit: 'Diamonds', number: 12 },
  { suit: 'Diamonds', number: 13 },
  { suit: 'Hearts', number: 1 },
  { suit: 'Hearts', number: 2 },
  { suit: 'Hearts', number: 3 },
  { suit: 'Hearts', number: 4 },
  { suit: 'Hearts', number: 5 },
  { suit: 'Hearts', number: 6 },
  { suit: 'Hearts', number: 7 },
  { suit: 'Hearts', number: 8 },
  { suit: 'Hearts', number: 9 },
  { suit: 'Hearts', number: 10 },
  { suit: 'Hearts', number: 11 },
  { suit: 'Hearts', number: 12 },
  { suit: 'Hearts', number: 13 },
];

export const HAND_RANK: { [key in Hand]: string } = {
  [Hand.Drop]: 'Drop',
  [Hand.HighCard]: 'High Card',
  [Hand.OnePair]: 'One Pair',
  [Hand.TwoPair]: 'Two Pair',
  [Hand.ThreeOfAKind]: 'Three of a Kind',
  [Hand.Straight]: 'Straight',
  [Hand.Flush]: 'Flush',
  [Hand.FullHouse]: 'Full House',
  [Hand.FourOfAKind]: 'Four of a Kind',
  [Hand.StraightFlush]: 'Straight Flush',
  [Hand.RoyalStraightFlush]: 'Royal Straight Flush',
};
