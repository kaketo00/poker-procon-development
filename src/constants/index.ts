import type { Suit } from '@/schema/card';

export const SERVICE_NAME = 'Flush Code Hack';

export const API_HOST = 'http://localhost:3000';

export const SUITS: Suit[] = ['Spades', 'Hearts', 'Clubs', 'Diamonds'];

export const LOG_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH:mm:ss:SSS';

export const DATA_FILE_PATH = 'data';

export const SEQUENCE_FILE_NAME = 'sequence.txt';

export const LOG_FILE_PATH = 'logs';

export const ALERT_CLOSE_DELAY = 2000;
