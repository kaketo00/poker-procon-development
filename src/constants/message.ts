import type { ErrorCode } from '@/schema/common';

export const SUCCESS_MESSAGES = {
  createGame: '新しいゲームを作成しました。',
  startGame: 'ゲームを開始しました。',
};

export const ERROR_MESSAGES: { [key in ErrorCode]: string } = {
  BA001: '存在しないAPIです。',

  BB001: '必須項目です。',
  BB002: '数値で入力してください。',
  BB003: '整数で入力してください。',
  BB004: '参加者を4人選択してください。',
  BB005: '1以上の値を入れてください。',
  BB006: '1,000以上の値を入れてください。',
  BB007: '1,000以下の値を入れてください。',
  BB008: '10,000以下の値を入れてください。',
  BB009: '1,000,000以下の値を入れてください。',

  BC001: 'ディレクトリの作成に失敗しました。',
  BC002: 'ディレクトリの読み込みに失敗しました。',
  BC003: 'ファイルの書き込みに失敗しました。',
  BC004: 'ファイルの読み込みに失敗しました。',

  BD001: 'ゲームが見つかりませんでした。',

  BZ001: '予期しないエラーが発生しました。',
};
