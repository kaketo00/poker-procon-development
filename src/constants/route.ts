/**
 * Routing
 */

const ROUTE = {
  S_A01_001: '/',
  S_B01_001: '/games',
  S_B01_002: '/games/:id',
  S_B01_003: '/games/:id/:round',
};

export default ROUTE;
