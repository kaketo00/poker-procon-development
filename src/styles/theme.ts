import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
