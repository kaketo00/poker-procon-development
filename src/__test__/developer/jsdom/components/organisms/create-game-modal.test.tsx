import { ChakraProvider } from '@chakra-ui/react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

import type { CreateGameModalProps } from '@/components/organisms/create-game-modal';
import CreateGameModal from '@/components/organisms/create-game-modal';
import { ERROR_MESSAGES, SUCCESS_MESSAGES } from '@/constants/message';
import theme from '@/styles/theme';

jest.mock('@chakra-ui/portal', () => ({
  Portal: jest.fn(({ children }: { children: ReactNode }) => children),
}));

jest.mock('@/clients/game', () => ({
  async createGame() {
    return {
      id: '1',
      message: SUCCESS_MESSAGES.createGame,
    };
  },
}));

const Component = (props: CreateGameModalProps) => (
  <ChakraProvider theme={theme}>
    <QueryClientProvider client={new QueryClient()}>
      <CreateGameModal {...props} />
    </QueryClientProvider>
  </ChakraProvider>
);

describe('CreateGameModal', () => {
  test('スナップショット', () => {
    const { asFragment } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={jest.fn()}
        setIsLoading={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('バリデーション/integer', async () => {
    const postCreateGame = jest.fn();
    const setIsLoading = jest.fn();

    const { getAllByText, getByRole } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={postCreateGame}
        setIsLoading={setIsLoading}
      />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'round' }), {
      target: {
        value: '0.1',
      },
    });
    await userEvent.click(getByRole('button', { name: '作成' }));

    await waitFor(() => {
      expect(getAllByText(ERROR_MESSAGES.BB003)).toHaveLength(1);
    });
  });

  test('バリデーション/最小値', async () => {
    const postCreateGame = jest.fn();
    const setIsLoading = jest.fn();

    const { getByRole, getAllByText } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={postCreateGame}
        setIsLoading={setIsLoading}
      />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'round' }), {
      target: {
        value: '0',
      },
    });
    await userEvent.click(getByRole('button', { name: '作成' }));

    await waitFor(() => {
      expect(getAllByText(ERROR_MESSAGES.BB005)).toHaveLength(1);
    });
  });

  test('バリデーション/最大値', async () => {
    const postCreateGame = jest.fn();
    const setIsLoading = jest.fn();

    const { getByRole, getAllByText } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={postCreateGame}
        setIsLoading={setIsLoading}
      />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'round' }), {
      target: {
        value: '10001',
      },
    });
    await userEvent.click(getByRole('button', { name: '作成' }));

    await waitFor(() => {
      expect(getAllByText(ERROR_MESSAGES.BB008)).toHaveLength(1);
    });
  });

  test('バリデーション/必須項目', async () => {
    const postCreateGame = jest.fn();
    const setIsLoading = jest.fn();

    const { getByRole, getByText } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={postCreateGame}
        setIsLoading={setIsLoading}
      />
    );

    await userEvent.click(getByRole('button', { name: '作成' }));

    await waitFor(() => {
      expect(getByText(ERROR_MESSAGES.BB004)).toBeInTheDocument();
    });
  });

  test('ゲーム作成', async () => {
    const postCreateGame = jest.fn();
    const setIsLoading = jest.fn();

    const { getByRole } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={jest.fn()}
        postCreateGame={postCreateGame}
        setIsLoading={setIsLoading}
      />
    );

    await userEvent.click(getByRole('checkbox', { name: 'player-0' }));
    await userEvent.click(getByRole('checkbox', { name: 'player-1' }));
    await userEvent.click(getByRole('checkbox', { name: 'player-2' }));
    await userEvent.click(getByRole('checkbox', { name: 'player-3' }));
    await userEvent.click(getByRole('button', { name: '作成' }));

    await waitFor(() => {
      expect(setIsLoading).toHaveBeenCalledWith(true);
      expect(postCreateGame).toHaveBeenCalled();
    });
  });

  test('モダールを閉じる', async () => {
    const onClose = jest.fn();

    const { getByRole } = render(
      <Component
        players={['Player1', 'Player2', 'Player3', 'Player4']}
        isOpen={true}
        onClose={onClose}
        postCreateGame={jest.fn()}
        setIsLoading={jest.fn()}
      />
    );

    await userEvent.click(getByRole('button', { name: 'Close' }));

    await waitFor(() => {
      expect(onClose).toHaveBeenCalledWith();
    });
  });
});
