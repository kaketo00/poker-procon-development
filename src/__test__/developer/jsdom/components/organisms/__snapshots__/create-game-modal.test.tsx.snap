// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGameModal スナップショット 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="chakra-modal__overlay css-hdd9l7"
    data-aria-hidden="true"
    style="opacity: 0;"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="chakra-modal__content-container css-1u2cvaz"
      tabindex="-1"
    >
      <section
        aria-describedby="chakra-modal--body-:r0:"
        aria-labelledby="chakra-modal--header-:r0:"
        aria-modal="true"
        class="chakra-modal__content css-fjdl7f"
        id="chakra-modal-:r0:"
        role="dialog"
        style="opacity: 0; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-modal__header css-9fgtzh"
          id="chakra-modal--header-:r0:"
        >
          ゲーム追加
        </header>
        <button
          aria-label="Close"
          class="chakra-modal__close-btn css-1ik4h6n"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-modal__body css-qlig70"
          id="chakra-modal--body-:r0:"
        >
          <form>
            <div
              class="css-18936vg"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="css-1yp4ln"
                >
                  <label
                    class="chakra-form__label css-la3zr9"
                    for="players"
                    id="field-:r3:-label"
                  >
                    プレイヤー
                  </label>
                  <div
                    class="css-2muyj4"
                  >
                    <label
                      class="chakra-checkbox css-ctlur2"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-label="player-0"
                        class="chakra-checkbox__input"
                        id="field-:r3:"
                        name="players"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Player1"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-1ydjfm6"
                      />
                      <span
                        class="chakra-checkbox__label css-6x44c9"
                      >
                        Player1
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-ctlur2"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-label="player-1"
                        class="chakra-checkbox__input"
                        id="field-:r3:"
                        name="players"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Player2"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-1ydjfm6"
                      />
                      <span
                        class="chakra-checkbox__label css-6x44c9"
                      >
                        Player2
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-ctlur2"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-label="player-2"
                        class="chakra-checkbox__input"
                        id="field-:r3:"
                        name="players"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Player3"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-1ydjfm6"
                      />
                      <span
                        class="chakra-checkbox__label css-6x44c9"
                      >
                        Player3
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-ctlur2"
                    >
                      <input
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-label="player-3"
                        class="chakra-checkbox__input"
                        id="field-:r3:"
                        name="players"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Player4"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-1ydjfm6"
                      />
                      <span
                        class="chakra-checkbox__label css-6x44c9"
                      >
                        Player4
                      </span>
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="css-k008qs"
                >
                  <label
                    class="chakra-form__label css-la3zr9"
                    for="round"
                    id="field-:r4:-label"
                  >
                    ラウンド数
                  </label>
                  <div
                    class="chakra-numberinput css-3e5t3k"
                  >
                    <input
                      aria-label="round"
                      aria-readonly="false"
                      aria-required="false"
                      aria-valuemax="9007199254740991"
                      aria-valuemin="-9007199254740991"
                      aria-valuenow="100"
                      aria-valuetext="100"
                      autocomplete="off"
                      autocorrect="off"
                      class="chakra-numberinput__field css-1g0mrv8"
                      id="round"
                      inputmode="decimal"
                      name="totalRound"
                      pattern="[0-9]*(.[0-9]+)?"
                      role="spinbutton"
                      type="text"
                      value="100"
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="css-k008qs"
                >
                  <label
                    class="chakra-form__label css-la3zr9"
                    for="initialPoint"
                    id="field-:r5:-label"
                  >
                    初期ポイント
                  </label>
                  <div
                    class="chakra-numberinput css-3e5t3k"
                  >
                    <input
                      aria-label="initialPoint"
                      aria-readonly="false"
                      aria-required="false"
                      aria-valuemax="9007199254740991"
                      aria-valuemin="-9007199254740991"
                      aria-valuenow="20000"
                      aria-valuetext="20000"
                      autocomplete="off"
                      autocorrect="off"
                      class="chakra-numberinput__field css-1g0mrv8"
                      disabled=""
                      id="initialPoint"
                      inputmode="decimal"
                      name="initialPoint"
                      pattern="[0-9]*(.[0-9]+)?"
                      role="spinbutton"
                      type="text"
                      value="20000"
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="css-k008qs"
                >
                  <label
                    class="chakra-form__label css-la3zr9"
                    for="fee"
                    id="field-:r6:-label"
                  >
                    参加フィー
                  </label>
                  <div
                    class="chakra-numberinput css-3e5t3k"
                  >
                    <input
                      aria-label="fee"
                      aria-readonly="false"
                      aria-required="false"
                      aria-valuemax="9007199254740991"
                      aria-valuemin="-9007199254740991"
                      aria-valuenow="200"
                      aria-valuetext="200"
                      autocomplete="off"
                      autocorrect="off"
                      class="chakra-numberinput__field css-1g0mrv8"
                      disabled=""
                      id="fee"
                      inputmode="decimal"
                      name="fee"
                      pattern="[0-9]*(.[0-9]+)?"
                      role="spinbutton"
                      type="text"
                      value="200"
                    />
                  </div>
                </div>
              </div>
              <div
                class="css-9idvp9"
              >
                <button
                  class="chakra-button css-11494p6"
                  type="reset"
                >
                  キャンセル
                </button>
                <button
                  class="chakra-button css-qqef0j"
                  type="submit"
                >
                  作成
                </button>
              </div>
            </div>
          </form>
        </div>
        <footer
          class="chakra-modal__footer css-k0waxj"
        >
          <div
            class="css-1vakbk4"
          />
        </footer>
      </section>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-label="Notifications-top"
    aria-live="polite"
    id="chakra-toast-manager-top"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
  />
  <div
    aria-label="Notifications-top-left"
    aria-live="polite"
    id="chakra-toast-manager-top-left"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-top-right"
    aria-live="polite"
    id="chakra-toast-manager-top-right"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-bottom-left"
    aria-live="polite"
    id="chakra-toast-manager-bottom-left"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-bottom"
    aria-live="polite"
    id="chakra-toast-manager-bottom"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
  />
  <div
    aria-label="Notifications-bottom-right"
    aria-live="polite"
    id="chakra-toast-manager-bottom-right"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
